physical/raft/raft.go

```diff
@@ -1622,10 +1617,26 @@ func (b *RaftBackend) applyLog(ctx context.Context, command *LogData) error {
 		resp = chunkedSuccess.Response
 	}
 
-	if resp, ok := resp.(*FSMApplyResponse); !ok || !resp.Success {
+	fsmar, ok := resp.(*FSMApplyResponse)
+	if !ok || !fsmar.Success {
 		return errors.New("could not apply data")
 	}
 
+	// populate command with our results
+	if fsmar.EntrySlice != nil {
```

Comment:
When would EntrySlice be nil?

Reply:
It wouldn't be, based on our code now, but it doesn't hurt to have a check for it, does it?

Reply:
Perhaps, perhaps not.  When I see an if like that, I'm forced to ask: what should the else case be?  You're protecting us against obvious failures (panic on nil) but possibly hiding a problem that could be harder to debug.  And a person reading the code is forced to wonder the same thing I did, and possibly go digging in the code to understand the scenario that gave rise to the if.

Reply:
I guess what I'm saying is: if it should never be nil, and you want to program defensively, then handle the nil case by returning an error.

Reply:
Fixed in https://github.com/hashicorp/vault/pull/17151/commits/bb5e0bd9a14255e1685c4a9bef463d0dd1dc4fe1